builtin/credential/cert/backend.go

```diff
@@ -63,6 +69,37 @@ func (b *backend) invalidate(_ context.Context, key string) {
 	}
 }
 
+func (b *backend) fetchCRL(ctx context.Context, storage logical.Storage, name string, crl *CRLInfo) error {
+	response, err := http.Get(crl.CDP.Url)
+	if err != nil {
+		return err
+	}
+	if response.StatusCode == http.StatusOK {
+		body, err := io.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+		certList, err := x509.ParseCRL(body)
+		if err != nil {
+			return err
+		}
+		crl.CDP.ValidUntil = certList.TBSCertList.NextUpdate
+		return b.setCRL(ctx, storage, certList, name, crl.CDP)
+	}
+	return fmt.Errorf("unexpected response code %d fetching CRL from %s", response.StatusCode, crl.CDP.Url)
+}
+
+func (b *backend) updateCRLs(ctx context.Context, req *logical.Request) error {
+	for name, crl := range b.crls {
+		if crl.CDP != nil && time.Now().After(crl.CDP.ValidUntil) {
+			if err := b.fetchCRL(ctx, req.Storage, name, &crl); err != nil {
+				return err
```

Comment:
I'd be happy to defer this to another PR that I can create, but I think we should catch these errors and continue iterating over the list in `b.crls`. I'd rather not stop refreshing all CRLs if we have one url that goes bad. Still would fail the function but after the for with the first error encountered perhaps?

Reply:
I agree, and worth changing.

Reply:
Fixed.